### PR TITLE
 User documentation for GPU integration with Pyccel

### PR DESCRIPTION
**Pull request by bauom**
This PR serves as a starting point for discussing the user documentation for GPU integration with Pyccel. The goal of the documentation is to help users understand and use Pyccel effectively when building applications that run on GPUs.

We would like to focus on the following topics in the documentation:

`device` and `kernel` decorators
`CUDA` Built-in variables

We welcome feedback and suggestions on these topics, as well as any other ideas you may have for improving the user documentation on GPU integration with Pyccel. Please share your thoughts on how we can make the documentation more useful and user-friendly.

### CODE REVIEW

File: developer_docs/GPU_getting_started.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Getting started GPU
+
+## Decorators
+
+  * Kernel
+
+    In Pyccel, the `@kernel` decorator is used to indicate that a function should be treated as a CUDA kernel. A CUDA kernel is a function that is executed on a GPU and is typically used to perform a parallel computation over a large dataset.
+
+    The `@kernel` decorator is used to indicate to Pyccel that the function should be compiled as a CUDA kernel, and that it can be launched on a GPU using the appropriate syntax. For example, if we have a function decorated with `@kernel` like this:
```
Comment (framdani) [inline developer_docs/GPU_getting_started.md:9]: There seems to be some repetition. Could we rephrase it to something like this, for example ?
```suggestion
By applying the @kernel decorator to a function, Pyccel recognizes it as a CUDA kernel that can be launched on a GPU using the appropriate syntax.
```

**Review by framdani — CHANGES_REQUESTED** [review]

File: developer_docs/GPU_getting_started.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+@kernel
+def my_kernel(x: 'float64[:]', y: 'float64[:]', out: 'float64[:]'):
+    i = cuda.threadIdx(0) + cuda.blockIdx(0) * cuda.blockDim(0)
+    if i >= x.shape[0]:
+        return
+    out[i] = x[i] + y[i]
+```
+
+The kernel uses the `cuda.threadIdx(0)`, `cuda.blockIdx(0)`, and `cuda.blockDim(0)` built-in variables to compute the global index of the current thread within the input arrays.
```
[the remark this answers is not in the document]
Comment (bauom) [reply developer_docs/GPU_getting_started.md:77]: yeah i think this should be added but in a way that doesn't suggest that using `cuda.threadIdx(dim)` where `dim=0` is related to the array size.

[the remark this answers is not in the document]
Comment (bauom) [reply developer_docs/GPU_getting_started.md:77]: If it is clear that that paragraph only describe the code above your change would make sense but it shouldn't mean that it is a rule on how to use the built-in variables.

**Comment by EmilyBourne** [comment]
This PR is labelled as developer docs, but it looks like user docs to me. Doesn't this describe how to translate GPU code rather than how to develop GPU related Pyccel code?

**Comment by bauom** [comment]
> This PR is labelled as developer docs, but it looks like user docs to me. Doesn't this describe how to translate GPU code rather than how to develop GPU related Pyccel code?

I didn't notice as I was writing it, do you think it better to add the specifics of the Pyccel classes in this one or have a user doc that have the stuff written here and just reference it in the developer docs.

**Comment by EmilyBourne** [comment]
> > This PR is labelled as developer docs, but it looks like user docs to me. Doesn't this describe how to translate GPU code rather than how to develop GPU related Pyccel code?
> 
> I didn't notice as I was writing it, do you think it better to add the specifics of the Pyccel classes in this one or have a user doc that have the stuff written here and just reference it in the developer docs.

Personally, I think this looks good as user docs. I don't see anything aimed at devs so I would just reference it from developer docs

**Comment by EmilyBourne** [comment]
@bauom Please migrate this PR to the `pyccel/pyccel-cuda` repo. You can transfer your branch to that repo using the following commands:
```
git remote add cuda git@github.com:pyccel/pyccel-cuda.git
git checkout GPU_ddocs
git push -u cuda GPU_ddocs
```

**Comment by bauom** [comment]
moved to https://github.com/pyccel/pyccel-cuda/pull/22
